rpc/addrlock.go

```diff
@@ -0,0 +1,38 @@
+package rpc
+
+import (
+	"sync"
+
+	"github.com/ethereum/go-ethereum/common"
+)
+
+// AddrLocker is a mutex structure used to avoid querying outdated account data
+type AddrLocker struct {
```

Comment:
You probably have considered this, but I think you could just query the account nonce and not worry about this. tl;dr on why you opted for this?

Reply:
Nah the mutex is so that in the time in between when the nonce is retrieved to when the tx is broadcasted, but yeah I'm not sure exactly necessary this is because the account's are queried from the node and that may just be the committed accounts and not include the mempool transitions